File: exemplo.php
<?php

    // Constantes
    define("TAPI_ID", "<chave_tapi>");
    define("TAPI_SECRET", "<segredo>");
    define("ENDPOINT_TRADE_PATH", "/tapi/v3/");
    define("ENDPOINT_TRADE_API", "https://www.mercadobitcoin.net" . ENDPOINT_TRADE_PATH);

    // Nonce
    $date = new DateTime();
    $tapi_nonce = $date->getTimestamp();

    // Parâmetros
    $params = array(
        'tapi_method' => 'get_account_info',
        'tapi_nonce' => $tapi_nonce
    );
    $params = http_build_query($params);

    // Gerar MAC
    $params_string = ENDPOINT_TRADE_PATH . '?' . $params;
    $tapi_mac = hash_hmac('sha512', $params_string, TAPI_SECRET);

    # Gerar cabeçalho da requisição
    $headers = array(
        'Content-Type: application/x-www-form-urlencoded',
        'TAPI-ID: '. TAPI_ID,
        'TAPI-MAC: '. $tapi_mac
    );

    // Realizar requisição POST
    $ch = curl_init();
    curl_setopt($ch, CURLOPT_URL, ENDPOINT_TRADE_API);
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true); 
    curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);
    curl_setopt($ch, CURLOPT_CUSTOMREQUEST, "POST");
    curl_setopt($ch, CURLOPT_POSTFIELDS, $params);
    curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);

    $response = curl_exec($ch);
    $json_response = json_decode($response);
    $info = curl_getinfo($ch);
    curl_close($ch);

    // Exibindo dados da resposta no console
    print_r($json_response);
?>

File: exemplo.cs
ServicePointManager.Expect100Continue = true;
ServicePointManager.SecurityProtocol = SecurityProtocolType.Tls12;

using System;
using System.Net;
using System.Text;
using System.IO;
using System.Web;
using System.Security.Cryptography;

class Chamada 
{
    public static void Main(string[] args) 
    {   
        // Constantes
        string mbTapiID = "<chave_tapi>";
	    string mbTapiSecret = "<segredo>";
	    string requestHost = "https://www.mercadobitcoin.net";
	    string requestPath = "/tapi/v3/";

        // Nonce
        int nonce = (Int32)(DateTime.UtcNow.Subtract(new DateTime(1970, 1, 1))).TotalSeconds;

        // Gerar MAC
        string message = "/tapi/v3/?tapi_method=get_account_info&tapi_nonce=" + nonce.ToString();
        byte[] postData = Encoding.UTF8.GetBytes(message);
        byte[] ms = Encoding.UTF8.GetBytes(mbTapiSecret);
        byte[] hash = new HMACSHA512(ms).ComputeHash(postData);
        string mac = BitConverter.ToString(hash).Replace("-", "").ToLower();

        // Parâmetros
        string paramsEncoded = "tapi_method=" + Uri.EscapeDataString("get_account_info");
            paramsEncoded += "&tapi_nonce=" + Uri.EscapeDataString(nonce.ToString());
        byte[] data = Encoding.UTF8.GetBytes(paramsEncoded);
        
        // Realizar requisição POST
        var request = (HttpWebRequest)WebRequest.Create(requestHost+requestPath); 
        request.Method = "POST";

        // Gerar cabeçalho da requisição
        request.Headers.Add("TAPI-ID", mbTapiID);
        request.Headers.Add("TAPI-MAC", mac);
        request.ContentType = "application/x-www-form-urlencoded";      
        request.ContentLength = data.Length;

        // Obter o stream com o conteúdo da requisição
        Stream stream = request.GetRequestStream();
        stream.Write(data, 0, data.Length);
        stream.Close();

        // Obter resposta
        var response = (HttpWebResponse)request.GetResponse();
        var responseString = new StreamReader(response.GetResponseStream()).ReadToEnd();
        
        // Exibindo dados da resposta no console
        Console.WriteLine(responseString);

        // Fechando a resposta
        response.Close();
    }
}
Chamada.Main(null);